python/ray/remote_function.py

```diff
@@ -239,15 +259,28 @@ def f():
         updated_options = ray_option_utils.update_options(default_options, task_options)
         ray_option_utils.validate_task_options(updated_options, in_options=True)
 
-        # only update runtime_env when ".options()" specifies new runtime_env
+        # Only update runtime_env and re-calculate serialized runtime env info when
+        # ".options()" specifies new runtime_env.
+        serialized_runtime_env_info = self._serialized_base_runtime_env_info
         if "runtime_env" in task_options:
             updated_options["runtime_env"] = parse_runtime_env(
```

Comment:
I think we should no longer populate updated_options["runtime_env"] ?

Reply:
We basically have two options:
- Keep `updated_options["runtime_env"]` updated, as we did in the past
- Remove `runtime_env` from `updated_options`

I choose the first method, becase `updated_options` corresponds to `task_options` and `default_options`, so people have expectation there's `runtime_env` in these options.

Reply:
hmm, ok since we need to support `bind` as well, where this optimization is not used